beacon-chain/rpc/eth/v1/validator/validator.go

```diff
@@ -5,43 +5,171 @@ import (
 )
 
 // GetAttesterDuties requests the beacon node to provide a set of attestation duties,
 // which should be performed by validators, for a particular epoch.
-func (vs *Server) GetAttesterDuties(ctx context.Context, req *ethpb.AttesterDutiesRequest) (*ethpb.AttesterDutiesResponse, error) {
-	return nil, errors.New("Unimplemented")
+func (vs *Server) GetAttesterDuties(ctx context.Context, req *v1.AttesterDutiesRequest) (*v1.AttesterDutiesResponse, error) {
+	ctx, span := trace.StartSpan(ctx, "validatorv1.GetAttesterDuties")
+	defer span.End()
+
+	if vs.SyncChecker.Syncing() {
+		return nil, status.Error(codes.Unavailable, "Syncing to latest head, not ready to respond")
+	}
+
+	cs := vs.TimeFetcher.CurrentSlot()
+	currentEpoch := helpers.SlotToEpoch(cs)
+	if req.Epoch > currentEpoch+1 {
+		return nil, status.Errorf(codes.InvalidArgument, "Request epoch %d can not be greater than next epoch %d", req.Epoch, currentEpoch+1)
+	}
+
+	s, err := vs.HeadFetcher.HeadState(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not get head state: %v", err)
+	}
+
+	vals := make([]iface.ReadOnlyValidator, len(req.Index))
+	for i, index := range req.Index {
+		val, err := s.ValidatorAtIndexReadOnly(index)
+		if _, ok := err.(*statev1.ValidatorIndexOutOfRangeError); ok {
+			return nil, status.Errorf(codes.InvalidArgument, "Invalid index: %v", err)
+		} else if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not get validator: %v", err)
+		}
+		vals[i] = val
+	}
+
+	// Advance state with empty transitions up to the requested epoch start slot.
+	// In case 1 epoch ahead was requested, we take the start slot of the current epoch.
+	// Taking the start slot of the next epoch would result in an error inside state.ProcessSlots.
+	var epochStartSlot types.Slot
+	if req.Epoch == currentEpoch+1 {
+		epochStartSlot, err = helpers.StartSlot(req.Epoch.Sub(1))
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not obtain epoch's start slot: %v", err)
+		}
+	} else {
+		epochStartSlot, err = helpers.StartSlot(req.Epoch)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not obtain epoch's start slot: %v", err)
+		}
+	}
+	if s.Slot() < epochStartSlot {
+		s, err = state.ProcessSlots(ctx, s, epochStartSlot)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not process slots up to %d: %v", epochStartSlot, err)
+		}
+	}
+
+	committeeAssignments, _, err := helpers.CommitteeAssignments(s, req.Epoch)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not compute committee assignments: %v", err)
+	}
+	var distinctCommitteeIndexes []types.CommitteeIndex
+assignmentLoop:
+	for _, assignment := range committeeAssignments {
+		for _, index := range distinctCommitteeIndexes {
+			if index == assignment.CommitteeIndex {
+				continue assignmentLoop
+			}
+		}
+		distinctCommitteeIndexes = append(distinctCommitteeIndexes, assignment.CommitteeIndex)
```

Comment:
This seems odd
1.) We could use a counter instead of list if it's just looking at slice length at the end
2.) `CommitteesAtSlot` is how many committees are in the slot right? In that case this is not correct